WSLGd: load keymap settings from .wslgconfig into weston.ini

fargiolas opened the pull request:
Allow user to customize weston settings by  providing a `.weston.ini` file in User Profile directory.

The idea is to give advanced users an easy way to configure a keymap to fallback to if weston rdp backend logic fails to find a mapping between current keyboard layout and XKB ones. See https://github.com/microsoft/weston-mirror/pull/144 for further details.

hideyukn88 commented:
Hi. we don't want to expose weston.ini directly from Windows side, this is why we have exposed only certain configuration via .wslgconfig. If the configuration of default keyboard is necessary, please expose it via .wslgconfig, thanks!

fargiolas committed:
WSLGd: import keyboard config into weston.ini

Introduce a new "weston-keyboard" section in .wslgconfig where you can
configure weston/xkb keymap settings that will be used as a fallback if
no xkb mapping can be found for current keyboard layout.

Syntax is the same of weston.ini(5). Only keymap related key-value
pairs are imported into WSLG system weston.ini, anything else is
ignored.

fargiolas commented:
@hideyukn88 see the new commit, it should address your concerns.

The idea is to add a new `[weston-keyboard]` (or whichever name you like) section in `.wslgconfig` where you can configure keymap settings that will be merged into wslg weston.ini when WSLGd starts and used as a fallback when keyboard layout mapping fails.

This way only keymap related settings are merged into weston.ini instead of exposing the full config file to the user. 

This obviously depends on https://github.com/microsoft/weston-mirror/pull/144 being merged as at the moment "us" layout is hardcoded in the rdp backend if layout mapping fails.

fargiolas commented:
@hideyukn88 any feedback on the latest proposed changes? now that the other PR has been merged (thanks!) it would definitely make life easier for us custom layout users

hideyukn88 commented:
@fargiolas, currently we are evaluating the idea to have /etc/wslg.conf, similar to WSL's /etc/wsl.conf, inside user-distro filesystem, so we could do this utilizing that, so please hold on, thanks!

fargiolas commented:
> @fargiolas, currently we are evaluating the idea to have /etc/wslg.conf, similar to WSL's /etc/wsl.conf, inside user-distro filesystem, so we could do this utilizing that, so please hold on, thanks!

That would be great! Thanks for the update!

cassiocsantana commented:
I'm looking forward to this update

Ckln commented:
Same 👀
